app/controllers/messages_controller.rb

```diff
@@ -19,28 +18,35 @@ def create
   end
 
   def show
-    if @message.file_contents.nil?
-      @message.destroy
+    @crypt = Crypt.find params[:id], key_param
+    @text  = @crypt.text # triggers CryptError, but used in view
```

Comment:
I'm not sure why this is necessary. Would it be possible for the views to just keep using `@message` and not know that they're dealing with a crypt rather than a message?